src/link_prober/LinkProber.cpp

```diff
@@ -591,4 +644,50 @@ void LinkProber::updateIcmpSequenceNo()
     addChecksumCarryover(&icmpHeader->checksum, mIcmpChecksum);
 }
 
+//
+// ---> findNextTlv
+//
+// Find next TLV to process in rxBuffer
+//
+size_t LinkProber::findNextTlv(size_t readOffset, size_t bytesTransferred)
+{
+    size_t tlvSize = 0;
+    if (readOffset + sizeof(Tlv) <= bytesTransferred) {
+        Tlv *tlvPtr = reinterpret_cast<Tlv *> (mRxBuffer.data() + readOffset);
+        tlvSize = (sizeof(Tlv) + ntohs(tlvPtr->length));
+    }
+    return tlvSize;
+}
+
+//
+// ---> appendTlvCommand
+//
+// Append TlvCommand to the end of txBuffer
+//
+size_t LinkProber::appendTlvCommand(Command commandType)
+{
+    assert(mTxPacketSize + sizeof(TlvCommand) <= MUX_MAX_ICMP_BUFFER_SIZE);
+    TlvCommand *tlvCommand = reinterpret_cast<TlvCommand *> (mTxBuffer.data() + mTxPacketSize);
+    tlvCommand->type = TlvCommand::tlvtype;
+    tlvCommand->length = htons(1);
+    tlvCommand->command = static_cast<uint8_t> (Command::COMMAND_SWITCH_ACTIVE);
```

Comment:
If it's always COMMAND_SWITCH_ACTIVE why do we need the function parameter?

Reply:
the parameter `commandType` has a default value as `COMMAND_SWITCH_ACTIVE`, would it be better to leave it in the function signature if there is more command type in the future?